internal/statestore/redis.go

```diff
@@ -82,22 +82,26 @@ func NewRedis(cfg config.View, redisURL string, maskedURL string) (Service, erro
 	redisConn := pool.Get()
 	defer handleConnectionClose(&redisConn)
 
-	_, err := redisConn.Do("SELECT", "0")
-	// Encountered an issue getting a connection from the pool.
-	if err != nil {
-		redisLogger.WithFields(logrus.Fields{
-			"cmd":   "SELECT 0",
-			"error": err.Error()}).Error("state storage connection error")
-		return nil, fmt.Errorf("cannot connect to Redis at %s, %s", maskedURL, err)
-	}
-
-	redisLogger.Info("Connected to Redis")
 	return &redisBackend{
 		redisPool: pool,
 		cfg:       cfg,
 	}, nil
 }
 
+// HealthCheck indicates if the database is reachable.
+func (rb *redisBackend) HealthCheck(ctx context.Context) error {
```

Comment:
Can you add the timeout settings to https://github.com/GoogleCloudPlatform/open-match/blob/master/internal/statestore/redis.go#L76? Otherwise, the healthcheck function will hang if redis is not ready yet.

```
		Dial: func() (redis.Conn, error) {
			return redis.DialURL(
				redisURL,
				redis.DialConnectTimeout(5*time.Second),
				redis.DialReadTimeout(3*time.Second),
			)
		},
```